src/eng_back.c

```diff
@@ -419,13 +429,20 @@ static X509 *pkcs11_load_cert(ENGINE_CTX *ctx, const char *s_slot_cert_id)
 	}
 
 	/* In several tokens certificates are marked as private. We use the pin-value */
-	if (tok->loginRequired && ctx->pin) {
```

Comment:
Note: This is one area I wasn't entirely sure if I made the right change or not. The code was conflicting with the comment below. It is here required that `ctx->pin` be non-null, but below, `PKCS11_login` is called with the expectation that it _may_ be null. The comment seemed more correct since a null pin should be allowed, so I took out this `ctx->pin != NULL`  check.